examples/platform_services/lib/main.dart

```diff
@@ -26,28 +25,42 @@ class _PlatformServicesState extends State<PlatformServices> {
             new Text('Hello from Flutter!'),
             new RaisedButton(
               child: new Text('Get Location'),
-              onPressed: _getLocation
+              onPressed: _requestLocation,
             ),
-            new Text('Latitude: $_latitude, Longitude: $_longitude'),
-          ]
-        )
-      )
+            new FutureBuilder<dynamic>(
+              future: _locationRequest,
+              builder: _buildLocation,
+            ),
+          ],
+        ),
+      ),
     );
   }
 
-  Future<Null> _getLocation() async {
-    final Map<String, String> message = <String, String>{'provider': 'network'};
-    final Map<String, dynamic> reply = await PlatformMessages.sendJSON('getLocation', message);
-    // If the widget was removed from the tree while the message was in flight,
-    // we want to discard the reply rather than calling setState to update our
-    // non-existent appearance.
-    if (!mounted)
-      return;
+  void _requestLocation() {
     setState(() {
-      _latitude = reply['latitude'].toDouble();
-      _longitude = reply['longitude'].toDouble();
+      _locationRequest = new PlatformMethodChannel('geo').invokeMethod(
```

Comment:
Is it recommend to make a new channel for each request?  If so, I wonder if we can make it const constructible.  No action needed.  Just wondering aloud.